legend-engine-test-runner-service/src/main/java/org/finos/legend/engine/testable/service/assertion/ServiceTestAssertionEvaluator.java

```diff
@@ -103,9 +107,10 @@ else if (result instanceof StreamingResult)
                 throw new RuntimeException(e);
             }
         }
+        // TODO: EqualToTDS
         else
         {
-            return null;
+            throw new UnsupportedOperationException("Unsupported assertion '" + testAssertion.id + "' of type '" + testAssertion.getClass().getName() + "'");
```

Comment:
I don't think we should throw as we expect to have multiple visitor and we should pick non-null values. So we should not throw exception here

Reply:
The issue with this is that we give a false positive and the test result comes back as `TestPass` if there is no implementation for that assertion type yet. We could have multiple visitors but after going through the core assertion types and all the extra ones if its still null then I think we should throw an error and treat it as a `TestError`.  

Faced this issue when running the test through studio and hadn't finished writing assertToTDS.

Reply:
Lets throw this error where we are filtering nuls